lib/model/message.dart

```diff
@@ -138,6 +138,36 @@ class MessageStoreImpl with MessageStore {
     }
   }
 
+  void _handleUpdateMessageEventMove(UpdateMessageEvent event) {
+    if (event.origTopic == null) {
+      // There was no move.
+      assert(() {
+        if (event.newStreamId != null && event.origStreamId != null
+            && event.newStreamId != event.origStreamId) {
+          // This should be impossible; `orig_subject` (aka origTopic) is
+          // documented to be present when either the stream or topic changed.
+          debugLog('Malformed UpdateMessageEvent: stream move but no origTopic'); // TODO(log)
+        }
+        return true;
+      }());
+      return;
+    }
+
+    if (event.newTopic == null) {
```

Comment:
I found this check a bit confusing (as well as the following one on `origStreamId`) before looking at the API doc. It could be helpful to spell out what we are expecting these checks (something that I can add in my PRs stacked atop this).

Reply:
Hmm yeah, these are a bit terse.

Would this version explain it?
```diff
@@ -139,6 +139,9 @@ class MessageStoreImpl with MessageStore {
   }
 
   void _handleUpdateMessageEventMove(UpdateMessageEvent event) {
+    // The interaction between the fields of these events are a bit tricky.
+    // For reference, see: https://zulip.com/api/get-events#update_message
+
     if (event.origTopic == null) {
       // There was no move.
       assert(() {
@@ -154,10 +157,12 @@ class MessageStoreImpl with MessageStore {
     }
 
     if (event.newTopic == null) {
-      assert(debugLog('Malformed UpdateMessageEvent: move but no topic')); // TODO(log)
+      // The `subject` field (aka newTopic) is documented to be present on moves.
+      assert(debugLog('Malformed UpdateMessageEvent: move but no newTopic')); // TODO(log)
       return;
     }
     if (event.origStreamId == null) {
+      // The `stream_id` field (aka origStreamId) is documented to be present on moves.
       assert(debugLog('Malformed UpdateMessageEvent: move but no origStreamId')); // TODO(log)
       return;
     }
```

(That also fixes "no topic" to "no newTopic" — that's from a draft where I'd renamed that field only to "topic" rather than "newTopic".)

Reply:
Yeah this works.